### ISSUE TITLE
Can't re-connect after error with OkHttp lib

### ISSUE THREAD

**Issue by stairs** [forresthopkinsa/StompProtocolAndroid#6]
OkHttp websocket ERROR event means that socket is closed and no further calls to listener will be made. However, next piece of code only notifies the client that connection is closed when CLOSE event is received.
```
    void emitLifecycleEvent(@NonNull LifecycleEvent lifecycleEvent) {
        Log.d(TAG, "Emit lifecycle event: " + lifecycleEvent.getType().name());
        mLifecycleStream.onNext(lifecycleEvent);
        if (lifecycleEvent.getType().equals(LifecycleEvent.Type.CLOSED))
            mConnectionStream.onNext(false);
    }
```
As a result we get "Attempted to connect when already connected" when socket is actually closed.

**Comment by stairs** [comment]
I'll create a PR with fast fix but I won't be able to test right now. @forresthopkinsa Please, take a look and test if you can.

**Comment by forresthopkinsa** [comment]
Absolutely, I'll check it out when I get to work in about an hour.

**Comment by forresthopkinsa** [comment]
You're totally right. With JWS, messages can still be sent after an Error event, whereas with OkHttp, a Failure is a combination of an Error and a Close. The code should definitely reflect that.

**Comment by stairs** [comment]
Thanks a lot for fixing this one! Unfortunately, I've found another issue after this one was fixed. I'll report it separately.

**Comment by souravbasu14** [comment]
Hi,
I faced the reconnect problem after n/w disconnet/connect in the upstream project. i put my issue there [ref: https://github.com/NaikSoftware/StompProtocolAndroid/issues/87] where i saw the same discussion and came to this downstream project.
I tried with 17.11.0 version but still facing the same reconnect problem.
Is it the proper version to be checked or I am missing something.

**Comment by forresthopkinsa** [comment]
@souravbasu14 That fix wasn't included in the 17.11.0 release. It was released a day later, and it's scheduled to be included in 17.12.0. You can, however, use a non-release version, like so:

``` gradle
dependencies {
    compile 'com.github.forresthopkinsa:StompProtocolAndroid:965f1275ad'
}
```

**Comment by souravbasu14** [comment]
@forresthopkinsa I did a quick verification with the above version. Below are my steps and observations:
1. Once the network is disconnected, I am getting ERROR state. From there I called reconnect().
2. On network connection, stompclient is getting connected and i receive OPEN state.
3. Old subscribed  topics were not restored. No data is coming after reconnect
4. After step 2, OPEN and ERROR goes into a loop. After ~1min interval i get a ERROR state though my network doesn't change. And as per Step 1, i called reconnect(). So i get OPEN state. And this cycle continues. Also previous subscribed topics are not restored.

Also observed a crash which leads the app to crash.  Attached is the crashlog which i received in crashlytics.
[crash.txt](https://github.com/forresthopkinsa/StompProtocolAndroid/files/1524336/crash.txt)
